ha/core/controllers/pcs/node_controller.py

```diff
@@ -242,12 +264,44 @@ def start(self, nodeid: str) -> dict:
             Log.error(f"{nodeid} status is {_node_status}, node may not be started.")
             raise ClusterManagerError(f"Failed to start {nodeid} as found unhandled status {_node_status}")
 
-class PcsHWNodeController(PcsVMNodeController):
+    @controller_error_handler
+    def stop(self, node_id: str, timeout: int= -1, **op_kwargs) -> dict:
+        """
+        Stop Node with nodeid.
+        Args:
+            nodeid (str): Node ID from cluster nodes.
+        Returns:
+            ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
+                status: Succeeded, Failed, InProgress
+        """
+        timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
+        node_status = self.nodes_status([node_id]).get(node_id)
+        if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
+            Log.info(f"For stop {node_id}, Node already in offline state.")
+            status = f"Node {node_id} is already in offline state."
+        elif node_status == NODE_STATUSES.POWEROFF.value:
+            raise ClusterManagerError(f"Failed to stop {node_id}. Node is in {node_status}.")
```

Comment:
Do We need an exception here ?

Reply:
Exception needed as POWEROFF is not supported for VM so this condition is not expected